src/optionsutils.nim

```diff
@@ -497,3 +497,37 @@ template optCmp*(self, cmp, value: untyped): untyped =
       if `cmp`(unsafeGet(a), unsafeGet(b)):
         return a
   )()
+
+import sequtils
+
+macro withSome*(body: untyped): untyped =
```

Comment:
Maybe rename `body` to `procDef` as it is the entire definition and not just a procedure body.